docs/contributing/git_cheatsheet.md

```diff
@@ -0,0 +1,515 @@
+-   **Nano**: `Ctrl+O`, then `Enter`, then `Ctrl+X`
+-   **VS Code**: save and close the editor
+
+> Multi-line commits are especially useful for giving extra context on why the change was made, or summarizing multiple related changes.
+
+To make this even easier, you can also use:
+
+<!-- run-disable -->
+
+```bash
+make commit
+```
+
+This gives you an interactive prompt to help you craft a properly formatted commit message. Super handy!
+
+Try to keep your commit messages clear and concise! Adhering to stdlib's [Git Style Guide][github-style-guide] ensures a well-structured and understandable commit history.
+
+## Pushing
+
+Once your branch is ready, you need to push your **local** changes to your forked repository on GitHub using:
+
+<!-- run-disable -->
+
+```bash
+git push
+```
+
+If this is the first time you’re pushing the branch, [Git][git] may prompt you to set an [upstream][git-remotes] branch. You can do this manually by running:
+
+<!-- run-disable -->
+
+```bash
+git push --set-upstream origin feature/my-new-feature
+```
+
+> **Note:** This is a **one-time setup** for this branch. After this, you can simply use `git push` for future updates.
+
+Once pushed, your changes will be available on GitHub, and you can proceed to create a pull request.
+
+## Pull Request
+
+A pull request (PR) is a way to suggest changes to a project. It lets maintainers review your work, give feedback, and approve the changes before adding them to the main repository. This is how you officially submit your contributions to `stdlib` after pushing your changes:
+
+1.  Go to your fork on GitHub.
+2.  Click "Compare & pull request."
+3.  Add a clear title and description (mention what you changed and why).
+
+Once you create your pull request, a review request will be **automatically** sent to the maintainers via `stdlib-bot`. Your code will then be reviewed, and you may need to make some tweaks before it gets merged.
+
+## Syncing
+
+The `develop` branch of your forked repository should always stay **identical** to the official `stdlib` repository. If you accidentally add commits to your local `develop` branch, you need to remove them before updating it.
+
+To update your local `develop` branch while making sure there are no unwanted changes, run:
+
+<!-- run-disable -->
+
+```bash
+git checkout develop
+git pull --ff-only upstream develop
+```
+
+> **Why use `--ff-only`?** This ensures your branch updates **only if no merge commits are needed**. If your `develop` branch has unexpected changes, this command will fail, alerting you that something is wrong.
+
+If the above pull fails because you accidentally made changes to `develop`, you can **reset** it to match the official repository:
+
+<!-- run-disable -->
+
+```bash
+git checkout develop
+git reset --hard upstream/develop
+```
+
+> **Warning:** This will delete any changes you made to `develop`. Make sure you don’t have important work in this branch before running this command.
```

Comment:
```suggestion
> **Warning:** This will delete any changes you made to `develop`. Make sure you don't have important work in this branch before running this command.
```